src/graph/optimizer/Optimizer.cpp

```diff
@@ -30,12 +30,35 @@ StatusOr<const PlanNode *> Optimizer::findBestPlan(QueryContext *qctx) {
   auto optCtx = std::make_unique<OptContext>(qctx);
 
   auto root = qctx->plan()->root();
-  auto status = prepare(optCtx.get(), root);
-  NG_RETURN_IF_ERROR(status);
-  auto rootGroup = std::move(status).value();
+  auto spaceID = qctx->rctx()->session()->space().id;
+
+  // auto status = preprocess(root, qctx, spaceID);
```

Comment:
We should figure out the relation between `prepare` and `preprocess`.

Reply:
I think `preprocess ` is operated on a raw plan, while `prepare` which invoked `convertToGroup` is to convert the raw plan to a group, which is used in the exploration phase.

The 3 phase of optimizer of TIDB:
```
// Phase 1: Preprocessing

// The target of this phase is to preprocess the plan tree by some heuristic
// rules which should always be beneficial, for example Column Pruning.

//------------------------------------------------------------------------------
// Phase 2: Exploration
//------------------------------------------------------------------------------
//
// The target of this phase is to explore all the logically equivalent
// expressions by exploring all the equivalent group expressions of each group.

//------------------------------------------------------------------------------
// Phase 3: Implementation
//------------------------------------------------------------------------------
//
// The target of this phase is to search the best physical plan for a Group
// which satisfies a certain required physical property.
```